blockstore/splitstore/markset.go

```diff
@@ -21,8 +21,19 @@ type MarkSet interface {
 	SetConcurrent()
 }
 
+type MarkSetVisitor interface {
+	MarkSet
+	ObjectVisitor
+}
+
 type MarkSetEnv interface {
+	// Create creates a new markset within the environment.
+	// name is a unique name for this markset, mapped to the filesystem in disk-backed environments
+	// sizeHint is a hint about the expected size of the markset
 	Create(name string, sizeHint int64) (MarkSet, error)
+	// CreateVisitor is like Create, but returns a wider interface that supports atomic visits.
```

Comment:
well, we need this if we want to keep bloom marksets around, as it's pointless to use `Visit` on them.

Reply:
So, what happens if I specify that I want to use a bloom filter? Does lotus start spamming my log with random errors?

Reply:
Basically, we need to better distinguish between "something went wrong" and "this isn't supported, move along". That can either be done with multiple interfaces/types, sentinel errors, or a boolean "supported".

Reply:
yeah, it will fail the compaction altogether. Perhaps we should kill the bloom code.

Reply:
Arguably, the best solution here is a `SupportsVisitor` method in the `MarkSetEnv` so that we can fail early (in constructor) if the user specifies the bloom filter for markset.

Reply:
I'd kill it if we're not using it. We can always revive it from git history.

Reply:
done and done -- have cake, eat it too.

So I added a `SupportVisitor` interface for future extensibility, and kill the unused bloom filter code.